code/modules/power/singularity/singularity.dm

```diff
@@ -296,6 +296,20 @@
 
 	step(src, movement_dir)
 
+/obj/singularity/proc/check_cardinals_range(steps, retry_with_move = FALSE)
+	. = length(GLOB.cardinals)			//Should be 4.
+	for(var/i in GLOB.cardinals)
+		. -= check_turfs_in(i, steps)	//-1 for each working direction
+	if(. && retry_with_move)			//If there's still a positive value it means it didn't pass. Retry with move if applicable
+		for(var/i in GLOB.cardinals)
+			if(step(src, i))			//Move in each direction.
```

Comment:
Uh, the previous proc didn't cause it to move.

Reply:
this is intended, during expansion it's called with retry_with_move to make it try to move to an open location so it doesn't get stuck on field walls.
the proc that it calls after the step is intended to NOT move it to prevent an infinite loop.